### DISCUSSION TITLE
Pre-placed vehicles: Migrate from game logic to EDEN layers

### DISCUSSION THREAD

**Discussion by Vdauphin** [Vdauphin/HeartsAndMinds#1263] [General]
Hey @Zakant and all,

I got some great feadback from https://github.com/Vdauphin/HeartsAndMinds/pull/1189
But apparently when the base start to be huge, lines are crossing everywhere and make the mission.sqm unreadable. Then I said, use the alternative method: drop in init field this `this call btc_veh_fnc_addVehFromSQM;`.

But then I thought about using [EDEN layers](https://community.bistudio.com/wiki/Eden_Editor:_Layer) to group vehicles: 
<img src="https://community.bistudio.com/wikidata/images/0/0c/A3_EdenEditor_CreateLayer.gif" width="800">
and use [getMissionLayerEntities](https://community.bistudio.com/wiki/getMissionLayerEntities) to catch them on mission start

![Sans titre](https://user-images.githubusercontent.com/14364400/146515476-ee15c881-1b3f-4491-b838-7bc230389e63.png)
![Desktop Screenshot 2021 12 17 - 09 36 28 53](https://user-images.githubusercontent.com/14364400/146515488-ecc69578-7842-441b-be2f-bc9a93e29304.png)

**What are you thinking about this?**

Cheers

**Comment by Zakant** [comment]
I think the layer thing is over-sophisticated. You can, however, use multiple game logics. The only thing they do is calling `_x call btc_db_fnc_add_veh;` on every synced vehicle. So there is no problem simply copy'n'pasting the logic. That way you can have "only logic per vehicle cluster".
Regarding the point that it makes the mission.sqm unreadable: Whoever tries to read that thing lost control over his life long ago ;)
No seriously: I don't see any usecase besides committing it to git or maybe removing old addons that would require you to ever take a look at that file.

**Reply by Vdauphin** [reply]
> I think the layer thing is over-sophisticated

Actually, I saw some people using it during their mission editing
I like also the fact that you can add to only one layer a vehicle

As you may notice, a lot of people struggle with the actual system. I want to make this one dumbfree.
If for exemple, a player remove the game logic, this will be a dead end for his edition. Deleting accidentally a layer, should not happen a lot because it is not directly on the map. And it is easy to fix by creating a new one with the correct name.

So I don't really know which approach is the best from a new user perspective

**Reply by Zakant** [reply]
Yeah sure, the layers are great. But I personally dont like to use them for anything more then "organizing" stuff. Adding functionality to them forces users to use them for exactly that purpose... 
The best / easiest way for new users would probably be to add a EDEN Attribute, althought thats not possible (afaik) from a mission.

**Reply by Vdauphin** [reply]
I see, I think I will take the risk. 
You will be allow to blame me if people complain or report bug!

**Reply by Zakant** [reply]
As long as you keep the option to use logics I'm happy. Shouldn't be that complicated to keep all methods available. Especially when the "layer" code also only calls `x call btc_db_fnc_add_veh;` for each vehicle in the layer.
Ohh and I most certainly will ;)

**Reply by Vdauphin** [reply]
Yea you are right
I will merge `btc_EDENinventory` and `btc_db_fnc_add_veh` and declare `btc_vehicles` in it
Then all approach should be easy to implement
